src/ui/AdvancedSearch/DateInput/BetweenDateInput.ts

```diff
@@ -31,20 +33,32 @@ export class BetweenDateInput extends DateInput {
   }
 
   public getValue(): string {
-    let firstDate = this.firstDatePicker.getValue();
-    let secondDate = this.secondDatePicker.getValue();
+    const firstDate = this.firstDatePicker.getDateValue();
```

Comment:
firstDateAsString is the date value as a string for the query,

firstDate is the date value as a actual date object to do timespan comparison (and check if the timespan is invalid)